ush/eva/gen_eva_obs_yaml.py

```diff
@@ -17,22 +17,34 @@ def gen_eva_obs_yaml(inputyaml, templateyaml, outputdir):
     except Exception as e:
         logging.error(f'Error occurred when attempting to load: {inputyaml}, error: {e}')
     # get just the observations part of the YAML
-    if 'observers' in jedi_yaml_dict['observations']:
-        jediobs = jedi_yaml_dict['observations']['observers']
+    if 'cost function' in jedi_yaml_dict:
+        # cost function is in marine DA var.yaml
+        if 'observations' in jedi_yaml_dict['cost function']:
+            jediobs = jedi_yaml_dict['cost function']['observations']['observers']
     else:
-        # the unit tests have a different YAML setup
-        jediobs = jedi_yaml_dict['observations']
+        # this is for atm DA var.yaml
+        if 'observers' in jedi_yaml_dict['observations']:
+            jediobs = jedi_yaml_dict['observations']['observers']
+        else:
+            # the unit tests have a different YAML setup
+            jediobs = jedi_yaml_dict['observations']
     # construct a simplified list of obsspaces for EVA
     evaobs = []
     for obsspace in jediobs:
         tmp_os = obsspace['obs space']
+        sim_var = tmp_os['simulated variables'][0]
```

Comment:
So just the first simulated variable? Why?

Reply:
There is only one variable in the list.  Use [0] only to get the variable name, not the list. Here is the output:
```
['absolute_dynamic_topography']
['sea_surface_temperature']
```